shell/core/plugin.ts

```diff
@@ -178,7 +178,8 @@ export class Plugin implements IPlugin {
     const nparts = name.split('/');
 
     // Support components in a sub-folder - component_name/index.vue (and ignore other componnets in that folder)
-    if (nparts.length === 2) {
+    // Allow store-scoped models via sub-folder - pkgname/models/storename/type will be registered as storename/type to avoid overwriting shell/models/type
+    if (nparts.length === 2 && type !== 'models') {
```

Comment:
We need to validate something similar isn't required when pkg'ing up harvester (see shell/pkg/auto-import). Will take a look today with https://github.com/rancher/dashboard/pull/6794